tianshou/data/batch.py

```diff
@@ -389,20 +201,13 @@ def __setitem__(self, index: Union[
             str, slice, int, np.integer, np.ndarray, List[int]],
             value: Any) -> None:
         """Assign value to self[index]."""
-        if isinstance(value, (list, tuple)):
-            value = np.asanyarray(value)
-        if isinstance(value, np.ndarray):
-            if not issubclass(value.dtype.type, (np.bool_, np.number)):
-                value = value.astype(np.object)
         if isinstance(index, str):
-            self.__dict__[index] = value
+            self.__setattr__(index, value)
             return
-        if not isinstance(value, (dict, Batch)):
-            if _is_batch_set(value):
-                value = Batch(value)
-            else:
-                raise TypeError("Batch does not supported value type "
-                                f"{type(value)} for item assignment.")
+        value = Batch(a=value).a
```

Comment:
There is no better way to do it ? it is not very elegant to create a unless field a and to extract it right after. Maybe it would be nice to create a new internal method use to cast value in Batch-compatible data type, that would be called both here and in __init__.

Reply:
Fixed in c7ed637 .